core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1817,6 +1842,9 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
             log.info("  <-pending");
 
         if (bestChain) {
+            boolean wasDead = dead.remove(txHash) != null;
```

Comment:
Resurrecting dead transactions is handled in the reorganize method, right? It basically just undoes the effect of each block on the old chain and then rolls forward on the new chain. So I thought that logic would handle this case.